src/meta_schedule/feature_extractor/per_block_feature.cc

```diff
@@ -624,53 +573,56 @@ void Feature::SubFeature::SetReuse(const LoopNest& loop_nest, int64_t top_loop_t
       });
     }
   }
-  // Step 3.2. Enumerate loops from inner to outer, find the first loop with reuse
-  int n_loops = loop_nest.loops.size();
-  const std::vector<const ForNode*>& loops = loop_nest.loops;
-  for (int i = n_loops - 1; i >= 0; --i) {
-    const ForNode* loop = loops[i];
-    // Case 1. Find an invariant loop, i.e. reuse with kLoopMultipleRead
+  // Default case: no reuse
+  ReuseType& reuse_type = this->reuse_type = ReuseType::kNoReuse;
+  double& reuse_dis_iter = this->reuse_dis_iter = 0;
+  double& reuse_dis_bytes = this->reuse_dis_bytes = 0;
+  int64_t& reuse_ct = this->reuse_ct = 0;
+  // Enumerate loops from inner to outer, find the first loop with reuse
+  for (int loop_idx = n_loops - 1; loop_idx >= 0; --loop_idx) {
+    const ForNode* loop = loops[loop_idx];
+    // Case 1. Found an invariant loop, which means the buffer is reused `n` times under this
+    // loop, where `n` is the loop extent. In this case, the reuse type is `kLoopMultipleRead`.
     if (!region_vars.count(loop->loop_var.get())) {
       reuse_type = ReuseType::kLoopMultipleRead;
+      // Step 1.1. Set `reuse_ct`
       if (const int64_t* extent = GetLoopIntExtent(loop)) {
         reuse_ct = *extent;
       } else {
         reuse_ct = 1;
       }
+      // Step 1.2. Set `reuse_dis_iter`
       reuse_dis_iter = 1;
-      for (int j = n_loops - 1; j > i; --j) {
-        if (const int64_t* extent = GetLoopIntExtent(loops[j])) {
+      for (int i = loop_idx + 1; i < n_loops; ++i) {
+        if (const int64_t* extent = GetLoopIntExtent(loops[i])) {
           reuse_dis_iter *= *extent;
         }
       }
+      // Step 1.3. Set `reuse_dis_bytes`
       reuse_dis_bytes = 0.0;
-      if (i == n_loops - 1) {
-        reuse_dis_bytes = top_loop_touch_bytes;
-      } else {
-        for (const auto& iter : buffer_touched_under_loop.at(loops[i + 1])) {
-          const BufferNode* buffer = iter.first;
-          const IntVec& numels = iter.second;
-          int64_t numel = std::accumulate(numels.begin(), numels.end(), int64_t(0));
-          reuse_dis_bytes += numel * buffer->dtype.bytes();
-        }
+      int buffer_idx = -1;
+      for (int64_t numel : buffer_touched_under_loop.at(loop_idx + 1)) {
+        const BufferNode* buffer = sub_features.at(++buffer_idx).buffer_;
+        reuse_dis_bytes += numel * buffer->dtype.bytes();
       }
       break;
     }
     // Case 2. Find serial reuse, i.e. reuse with kSerialMultipleReadWrite
-    const IntVec& touched = buffer_touched_under_loop.at(loop).at(buffer);
-    if (touched.size() >= 2) {
+    if (this->regions.size() >= 2) {
+      reuse_ct = this->regions.size() - 1;
```

Comment:
I guess it should be touch.size() - 1?

Reply:
they are the same in our current implementation

Reply:
`this->regions.size()` is the number of times a certain buffer is touched under a block, where `touch.size()` means the number of buffers touched.